### PR TITLE
storcon: deny external node configuration if an operation is ongoing

### PR DESCRIPTION
**Pull request by koivunej**
Per #8674, disallow node configuration while drain/fill are ongoing. Implement it by adding a only-http wrapper `Service::external_node_configure` which checks for operation existing before configuring.

Additionally:
- allow cancelling drain/fill after a pageserver has restarted and transitioned to WarmingUp

Fixes: #8674

### CODE REVIEW

File: test_runner/regress/test_storage_controller.py [hunk 1]
```diff
@@ -2091,3 +2091,44 @@ def test_storage_controller_step_down(neon_env_builder: NeonEnvBuilder):
...
+    env = neon_env_builder.init_start()
+
+    # make sure there is no secondary, so we will have something to drain
+    env.storage_controller.tenant_policy_update(env.initial_tenant, {"placement": {"Attached": 1}})
+    env.storage_controller.reconcile_until_idle()
+
+    first, _ = env.pageservers
+
+    # attached_at = int(env.storage_controller.locate(env.initial_tenant)[0]["node_id"])
+    # secondary_at = next(filter(lambda ps: ps.id != attached_at, env.pageservers))
```
Comment (koivunej) [inline test_runner/regress/test_storage_controller.py:2108]: does this test break if storage controller decided to put the attached on the second node?

File: test_runner/regress/test_storage_controller.py [hunk 2]
```diff
@@ -2091,3 +2091,44 @@ def test_storage_controller_step_down(neon_env_builder: NeonEnvBuilder):
...
+        return first_is("Draining")
+
+    env.storage_controller.configure_failpoints(("sleepy-drain-loop", "return(10000)"))
+    env.storage_controller.node_drain(first.id)
+
+    wait_until(10, 0.5, first_is_draining)
+
+    # restarting or the re-attach request from ps does not cancel drain
+    # but it makes it so that it cannot be cancelled from the api, bug?
+    first.restart()
```
Comment (koivunej) [inline test_runner/regress/test_storage_controller.py:2125]: asked about this in DMs, can move conversation here.

I think it should be possible to cancel it while it exists, regardless of state.

Comment (VladLazar) [reply test_runner/regress/test_storage_controller.py:2125]: yeah. good point. Let's remove the check [here](https://github.com/neondatabase/neon/blob/main/storage_controller/src/service.rs#L5036-L5040) and it's fill counterpart as discussed

Comment (koivunej) [reply test_runner/regress/test_storage_controller.py:2126]: done in 3f1cb7b2aafb3920d06514ad45f85a161d5f14fa

File: test_runner/regress/test_storage_controller.py [hunk 3]
```diff
@@ -2091,3 +2091,44 @@ def test_storage_controller_step_down(neon_env_builder: NeonEnvBuilder):
...
     )
+
+def test_storage_controller_ps_restarted_during_drain(neon_env_builder: NeonEnvBuilder):
+    # single unsharded tenant, two locations
+    neon_env_builder.num_pageservers = 2
+
+    env = neon_env_builder.init_start()
+
+    # make sure there is no secondary, so we will have something to drain
+    env.storage_controller.tenant_policy_update(env.initial_tenant, {"placement": {"Attached": 1}})
```
[the remark this answers is not in the document]
Comment (koivunej) [reply test_runner/regress/test_storage_controller.py:2103]: Yeah, was playing around with this, but I guess all of my learnings got mixed up with the problems caused by accidentally forbidding internal node re-configuring :)

48f048f0f

File: test_runner/regress/test_storage_controller.py (hunk 1 above)
Comment (koivunej) [reply test_runner/regress/test_storage_controller.py]: done in b0dcf3ddd

**Review by VladLazar — APPROVED** [review]

**Pull request merged**